File: HOT100/HOT100p4/35.搜索插入位置.py
#
# @lc app=leetcode.cn id=35 lang=python3
#
# [35] 搜索插入位置
#
from mytools import *
# @lc code=start
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while l <= r:
            mid = l + r >> 1
            if nums[mid] == target: return mid
            elif nums[mid] > target: r = mid - 1
            else: l = mid + 1
        return l
# @lc code=end

File: 1.数组/2.二分查找/35.搜索插入位置.cpp
/*
 * @lc app=leetcode.cn id=35 lang=cpp
 *
 * [35] 搜索插入位置
 */

// @lc code=start
class Solution {
    public:
        int searchInsert(vector<int>& nums, int target) {
            int l = 0, r = nums.size() - 1;
            while (l <= r)
            {
                int mid = l + r >> 1;
                if (nums[mid] > target)
                {
                    r = mid - 1;
                }
                else if (nums[mid] < target)
                {
                    l = mid + 1;
                }
                else
                {
                    return mid;
                }
            }
            return l;
        }
    };
// @lc code=end

